mne/time_frequency/tests/test_psd.py

```diff
@@ -14,99 +16,105 @@
-    assert_true(psds.shape == (len(picks), len(freqs)))
-    assert_true(np.sum(freqs < 0) == 0)
-    assert_true(np.sum(psds < 0) == 0)
-
-    n_fft = 2048  # the FFT size (n_fft). Ideally a power of 2
-    psds, freqs = compute_raw_psd(raw, tmin=tmin, tmax=tmax, picks=picks,
-                                  fmin=fmin, fmax=fmax, n_fft=n_fft, n_jobs=1,
-                                  proj=False)
-    psds_proj, freqs = compute_raw_psd(raw, tmin=tmin, tmax=tmax, picks=picks,
-                                       fmin=fmin, fmax=fmax, n_fft=n_fft,
-                                       n_jobs=1, proj=True)
-
-    assert_array_almost_equal(psds, psds_proj)
-    assert_true(psds.shape == (len(picks), len(freqs)))
-    assert_true(np.sum(freqs < 0) == 0)
-    assert_true(np.sum(psds < 0) == 0)
-
-
-@requires_version('scipy', '0.12')
-def test_psd_epochs():
-    """Test PSD estimation on epochs
-    """
-    raw = io.Raw(raw_fname)
-
-    exclude = raw.info['bads'] + ['MEG 2443', 'EEG 053']  # bads + 2 more
-
-    # picks MEG gradiometers
-    picks = pick_types(raw.info, meg='mag', eeg=False, stim=False,
-                       exclude=exclude)
-
-    picks = picks[:2]
-
-    n_fft = 512  # the FFT size (n_fft). Ideally a power of 2
-
-    tmin, tmax, event_id = -0.5, 0.5, 1
-    include = []
-    raw.info['bads'] += ['MEG 2443']  # bads
 
     # picks MEG gradiometers
     picks = pick_types(raw.info, meg='grad', eeg=False, eog=True,
-                       stim=False, include=include, exclude='bads')
-
+                       stim=False, include=include, exclude=exclude)
+
+    # -- Raw --
+    picks_psd = picks[:2]
+    kws_psd = dict(tmin=tmin, tmax=tmax, fmin=fmin, fmax=fmax,
+                   picks=picks_psd)  # Common to all
+    kws_welch = dict(n_fft=n_fft)
+    kws_mt = dict(low_bias=True)
+    funcs = [(psd_welch, kws_welch),
+             (psd_multitaper, kws_mt),
+             (compute_raw_psd, kws_welch)]
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        for func, kws in funcs:
+            kws = kws.copy()
+            kws.update(kws_psd)
+            kws.update(tmin=tmin, tmax=tmax)
+            psds, freqs = func(raw, proj=False, **kws)
+            psds_proj, freqs_proj = func(raw, proj=True, **kws)
+
+            assert_true(psds.shape == (len(kws['picks']), len(freqs)))
+            assert_true(np.sum(freqs < 0) == 0)
+            assert_true(np.sum(psds < 0) == 0)
+
+            # Make sure the projection doesn't change channels it shouldn't
+            assert_array_almost_equal(psds, psds_proj)
+            # Array input shouldn't work
+            assert_raises(ValueError, func, raw[:3, :20][0])
+        assert_true(len(w), 3)
+
+    # -- Epochs/Evoked --
     events = read_events(event_fname)
-
+    tmin, tmax, event_id = -0.5, 0.5, 1
     epochs = Epochs(raw, events[:10], event_id, tmin, tmax, picks=picks,
```

Comment:
for these examples you could use an EpochsArray putting sinusoidal functions in data without known amplitude. So you could test that the psd peaks are correct. Maybe it will also speed up the tests.